os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -159,27 +118,53 @@ bool binary_manager_scan_kbin(void)
 	kernel_binary_header_t header_data;
 	char filepath[BINARY_PATH_LEN];
 
-	ret = binary_manager_scan_bootparam();
-	if (ret == BINMGR_OK) {
-		bp_data = binary_manager_get_bpdata();
-		/* Verify running kernel binary based on bootparam */
-		snprintf(filepath, BINARY_PATH_LEN, BINMGR_DEVNAME_FMT, kernel_info.part_num[bp_data->active_idx]);
-		ret = binary_manager_verify_kernel_binary(filepath, &header_data);
-		if (ret == OK) {
-			/* Update inuse index and kernel version */
-			kernel_info.inuse_idx = bp_data->active_idx;
-			kernel_info.version = header_data.version;
-			bmvdbg("Kernel version [%u] %u\n", kernel_info.inuse_idx, kernel_info.version);
-			return true;
-		}
-	} else {
-		lldbg("ERROR!! Failed to scan boot param.\n");
-		lldbg("Please check whether the partition 'bootparam' exists in CONFIG_FLASH_PART_TYPE with 8192K size.\n");
+	bp_data = binary_manager_get_bpdata();
+	/* Verify running kernel binary based on bootparam */
+	snprintf(filepath, BINARY_PATH_LEN, BINMGR_DEVNAME_FMT, kernel_info.part_info[bp_data->active_idx].devnum);
+	ret = binary_manager_read_header(BINARY_KERNEL, filepath, &header_data, false);
```

Comment:
As modified you said, `binary_manager_read_header` returns not `OK`.
Please change the check condition.